### PR TITLE
fix: add early check for invalid `toMetricPrefixString` input

### PR DESCRIPTION
**Pull request by tbeltzun**
Fix https://github.com/GEOS-DEV/GEOS/issues/2651.

Use [`fpclassify`](https://en.cppreference.com/w/cpp/numeric/math/fpclassify) to avoid `-Werror=float-equal`.

### CODE REVIEW

**Comment by rrsettgast** [comment]
@tbeltzun can you post what the output looks like with the zero case?

**Comment by castelletto1** [comment]
Is this ready to be merged? It seems quite a minor fix

**Comment by paveltomin** [comment]
@rrsettgast simple fix, merge?

File: src/coreComponents/codingUtilities/StringUtilities.cpp [hunk 1]
```diff
@@ -73,6 +73,9 @@ string removeStringAndFollowingContent( string const & str,
 template< typename T >
 string toMetricPrefixString( T const & value )
 {
+  if( std::fpclassify( value ) == FP_ZERO )
+    return " 0.0  ";
+
```
Comment (wrtobin) [inline src/coreComponents/codingUtilities/StringUtilities.cpp]:
```suggestion
  if( std::fpclassify( value ) == FP_ZERO )
  {
    return " 0.0  ";
  }

```

**Review by rrsettgast — APPROVED** [review]

**Pull request merged**